desc/compute/utils.py

```diff
@@ -333,13 +333,26 @@ def get_transforms(keys, obj, grid, jitable=False, **kwargs):
     transforms = {"grid": grid}
     for c in derivs.keys():
         if hasattr(obj, c + "_basis"):
-            transforms[c] = Transform(
-                grid,
-                getattr(obj, c + "_basis"),
-                derivs=derivs[c],
-                build=True,
-                method=method,
-            )
+            basis = getattr(obj, c + "_basis")
+            # first check if we already have a transform with a compatible basis
+            for transform in transforms.values():
+                if basis.eq(getattr(transform, "basis", None)):
+                    ders = np.unique(
+                        np.vstack([derivs[c], transform.derivatives]), axis=0
+                    )
+                    # don't build until we know all the derivs we need
+                    transform.change_derivatives(ders, build=False)
+                    c_transform = transform
+                    break
+            else:  # if we didn't exit the loop early
+                c_transform = Transform(
+                    grid,
+                    basis,
+                    derivs=derivs[c],
+                    build=False,
+                    method=method,
+                )
+            transforms[c] = c_transform
         elif c == "B":
```

Comment:
c is a key in the dictionary derivs which sounds like a dictionary of derivatives. c corresponds to the magnetic field "B". Why is the magnetic field in the dictionary containing derivatives?

Confusing naming convention. Add a few comments in the code to clarify.

Reply:
c is meant to be just a quantity for which a transform exists, like R or Z or L. B is a transform that may exist as well (used for the boozer transform stuff). The `derivs` is a dictionary whose keys are things for which trasnforms exist, and that we may want derivs of, so I think it is natural for `"B"` to be here in its keys. `derivs` dict is explained by the docstring of `get_derivs` (https://github.com/PlasmaControl/DESC/blob/70001bcc9bc38c4e23d7d457ce5f9bb4898741ed/desc/compute/utils.py#L184)

Reply:
"B" here refers to the transform thats needed to fit the boozer harmonics

Reply:
I thought you meant the derivatives needed for the magnetic field. I'd still stay it's confusing to people who have used something like eq.compute("B"). Maybe add a comment in front of the elif in 356 saying "B means Boozer"?